pkg/fileserver/fs9p/shares.go

```diff
@@ -0,0 +1,42 @@
+package fs9p
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/sirupsen/logrus"
+)
+
+type Mount struct {
+	Listener net.Listener
+	Path     string
+}
+
+func StartShares(plan9Mounts []Mount) (defErr error) {
+	for _, m := range plan9Mounts {
+		server, err := New9pServer(m.Listener, m.Path)
+		if err != nil {
+			return fmt.Errorf("serving directory %s on %s: %w", m.Path, m.Listener.Addr().String(), err)
+		}
+		defer func() {
+			if defErr != nil {
+				if err := server.Stop(); err != nil {
+					logrus.Errorf("Error stopping 9p server: %v", err)
+				}
+			}
+		}()
```

Comment:
why do we need this defer block, it seems `defErr` value is not being set in the function?